src/java/org/apache/cassandra/locator/AbstractCloudMetadataServiceConnector.java

```diff
@@ -18,122 +18,30 @@
 
 package org.apache.cassandra.locator;
 
-import java.io.DataInputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
-import java.net.URISyntaxException;
-import java.net.URL;
-import java.util.Map;
-
-import com.google.common.collect.ImmutableMap;
-
-import org.apache.cassandra.config.CassandraRelevantProperties;
-import org.apache.cassandra.config.DurationSpec;
-import org.apache.cassandra.exceptions.ConfigurationException;
-
 import static java.lang.String.format;
-import static java.nio.charset.StandardCharsets.UTF_8;
 
-abstract class AbstractCloudMetadataServiceConnector
+abstract class AbstractCloudMetadataServiceConnector extends HttpServiceConnector
 {
     static final String METADATA_URL_PROPERTY = "metadata_url";
     static final String METADATA_REQUEST_TIMEOUT_PROPERTY = "metadata_request_timeout";
     static final String DEFAULT_METADATA_REQUEST_TIMEOUT = "30s";
```

Comment:
Should we remove the constant due to the `HttpSeedProvider#DEFAULT_REQUEST_TIMEOUT` existence?

Reply:
I mean probably the timeout should be a part of the `HttpServiceConnector` configuration.

Reply:
Not only that ... I went a little bit wild here for seeds and it is possible to specify headers from them in the configuration for `HttpSeedProvider` too but it would be cool if it was possible to specify headers for snitches as well. That would require to move headers logic in `HttpSeedProvider` to `HttpServiceConnector`.